core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -306,8 +311,15 @@ protected SqlValidatorImpl(
       SqlValidatorCatalogReader catalogReader,
       RelDataTypeFactory typeFactory,
       Config config) {
-    this.opTab = Objects.requireNonNull(opTab);
     this.catalogReader = Objects.requireNonNull(catalogReader);
+    SqlOperatorTable readerTable = (SqlOperatorTable) catalogReader;
+    if (opTab instanceof ChainedSqlOperatorTable) {
```

Comment:
Rather than having this instanceof check and all of the casting logic here, we could handle this in ChainedSqlOperatorTable's `of()` method. Any operator tables that are specified in `of()` which are themselves of type ChainedSqlOperatorTable would have their individual tables extracted and added.

Reply:
As far as I can tell this is not the current behavior of the `of()` method. Are you suggesting that I should add this behaviour?

Reply:
Yep, that's what I'm suggesting. It seems like a caller could want to chain ChainedSqlOperatorTable's using the of() method for other use cases too.